device/samples/NetStandardMultiplexSample/Program.cs

```diff
@@ -0,0 +1,138 @@
+            {
+                Console.WriteLine("Error in sample: {0}", ex.Message);
+            }
+
+            Console.WriteLine("Exited!\n");
+        }
+
+        static async Task MultiplexConnection()
+        {
+            for (int i = 0; i < 4; i++)
+            {
+                var auth = new DeviceAuthenticationWithRegistrySymmetricKey(deviceIds[i], devicePrimaryKeys[i]);
+                deviceClients[i] = DeviceClient.Create(
+                    IoTHubHostName,
+                    auth,
+                    new ITransportSettings[]
+                    {
+                        //Must specify Amqp_WebSocket_Only or Amqp_Tcp_Only
+                        new AmqpTransportSettings(TransportType.Amqp_WebSocket_Only)
+                        {
+                            //By default, connection pooling is turned off. It must be turned on explicitly using the below settings.
+                            AmqpConnectionPoolSettings = new AmqpConnectionPoolSettings()
+                            {
+                                Pooling = true,
+                                //If you would like to have only one TCP Connection set MaxPoolSize=1
+                                MaxPoolSize = 2
+                            }
+                        }
+                    });
+                await deviceClients[i].OpenAsync();
+            }
+
+            SendEvent().Wait();
+            ReceiveCommands().Wait();
+        }
+
+        static async Task SendEvent()
+        {
+            string dataBuffer;
+
+            Console.WriteLine("Devices sending {0} messages to IoTHub...\n", MESSAGE_COUNT);
+
+            for (int count = 0; count < MESSAGE_COUNT; count++)
+            {
+                for (int i = 0; i < MUX_DEVICES; i++)
+                {
+                    temperature = rnd.Next(20, 35);
+                    humidity = rnd.Next(60, 80);
+                    dataBuffer = string.Format("{{\"deviceId\":\"{0}\",\"messageId\":{1},\"temperature\":{2},\"humidity\":{3}}}", deviceIds[i], count, temperature, humidity);
+                    Message eventMessage = new Message(Encoding.UTF8.GetBytes(dataBuffer));
+                    eventMessage.Properties.Add("temperatureAlert", (temperature > TEMPERATURE_THRESHOLD) ? "true" : "false");
+                    Console.WriteLine("\t{0}> Sending message: {1}, Data: [{2}]", DateTime.Now.ToLocalTime(), count, dataBuffer);
+                    try
+                    {
+                        await deviceClients[i].SendEventAsync(eventMessage);
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine("Error in sending message: {0}", ex.Message);
+                    }
+                    Thread.Sleep(1000);
+                }
+            }
+        }
+
+        static async Task ReceiveCommands()
+        {
+            Console.WriteLine("\nDevices waiting for commands from IoTHub...\n");
```

Comment:
I'd add something "Press Ctrl+C" to stop and describe a way for someone to use either DeviceExplorer or the Azure Portal to send some commands to the devices. Otherwise someone running this sample could think that the app is hanging.